src/overlays/actors/ovl_En_Drs/z_en_drs.c

```diff
@@ -30,36 +28,98 @@ const ActorInit En_Drs_InitVars = {
-static CollisionCheckInfoInit2 D_80C1E60C = { 0, 0, 0, 0, MASS_IMMOVABLE };
-
-#endif
-
-extern ColliderCylinderInit D_80C1E5E0;
-extern CollisionCheckInfoInit2 D_80C1E60C;
-
-extern UNK_TYPE D_06000E70;
-extern UNK_TYPE D_06005A78;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Drs/func_80C1E290.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Drs/func_80C1E2D4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Drs/func_80C1E3DC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Drs/EnDrs_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Drs/EnDrs_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Drs/EnDrs_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Drs/func_80C1E4B0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Drs/func_80C1E568.s")
+static CollisionCheckInfoInit2 sColChkInfoInit = { 0, 0, 0, 0, MASS_IMMOVABLE };
+
+static AnimationInfoS sAnimations = { &gWeddingDressMannequinIdleAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 };
+
+void EnDrs_CollisionUpdate(EnDrs* this, GlobalContext* globalCtx) {
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
+
+void EnDrs_Setup(EnDrs* this, GlobalContext* globalCtx) {
+    s32 pad;
+    SkelAnime* temp = &this->skelAnime;
+
+    if ((this->moonMaskObjBankIndex >= 0) && func_8013D8DC(this->moonMaskObjBankIndex, globalCtx)) {
+        ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
+        SkelAnime_InitFlex(globalCtx, temp, &gWeddingDressMannequinSkel, NULL, this->jointTable, this->morphTable,
+                           WEDDING_DRESS_MANNEQUIN_MAX_LIMB);
+        SubS_ChangeAnimationByInfoS(temp, &sAnimations, 0);
+        Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+        CollisionCheck_SetInfo2(&this->actor.colChkInfo, DamageTable_Get(0x16), &sColChkInfoInit);
+        Actor_SetScale(&this->actor, 0.01f);
+        this->actor.draw = EnDrs_Draw;
+        this->actionFunc = EnDrs_Idle;
+    }
+}
+
+void EnDrs_Idle(EnDrs* this, GlobalContext* globalCtx) {
+}
+
+void EnDrs_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDrs* this = THIS;
+    this->moonMaskObjBankIndex = func_8013D924(OBJECT_MSMO, globalCtx);
+    this->actionFunc = EnDrs_Setup;
+}
+
+void EnDrs_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnDrs* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnDrs_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnDrs* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    if (this->actor.draw) {
+        SkelAnime_Update(&this->skelAnime);
+        EnDrs_CollisionUpdate(this, globalCtx);
+    }
+}
+
+void EnDrs_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnDrs* this = THIS;
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
```

Comment:
Is this required? It doesn't seem to be used at all.

Reply:
It can match with either gfxCtx or using globalCtx2, is one preferable over the other?

Reply:
empty pads don't have any effect

Reply:
We came to the conclusion that this is a globalCtx2 in Discord.